### PR TITLE
Upgrade Gutenberg [MAILPOET-3154]

### PR DESCRIPTION
**Pull request by pavel-mailpoet**
[MAILPOET-3154]

[MAILPOET-3154]: https://mailpoet.atlassian.net/browse/MAILPOET-3154

### CODE REVIEW

File: assets/js/src/form_editor/store/reducers/toggle_sidebar_panel.jsx [hunk 1]
```diff
@@ -21,11 +21,11 @@ const getRequiredAction = (openedPanels, panelId, toggleTo) => {
  * @return {object} Modified state object
  */
 export default (state, action) => {
-  if (action.toggleTo !== undefined && !['opened', 'closed'].includes(action.toggleTo)) {
-    throw new Error(`Unexpected toggleTo value "${action.toggleTo}"`);
-  }
+  let toggleTo;
```
Comment (costasovo) [inline assets/js/src/form_editor/store/reducers/toggle_sidebar_panel.jsx:24]: Just wondering, why this change had to be done?

**Review by costasovo — CHANGES_REQUESTED** [review]
I tried to use the editor created a form from a template and noticed a few issues:

1. When  I create a form from a template and click on a Heading block which is already in the form there is no font size selected in typography selection. Maybe they changed heading styles attributes and we need to update our mapping. Or maybe they dropped custom font size for heading :( I can see that post editor with Gutenberg 9.1 still has the custom size input for headings and paragraphs so I guess we need to activate it somehow maybe in block settings or in font size settings...

2. There is a warning in console when editor starts:
```
Warning: React does not recognize the `setRef` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `setref` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in h1 (created by Autocomplete)
    in Autocomplete (created by WithInstanceId(Autocomplete))
    in WithInstanceId(Autocomplete) (created by WithSpokenMessages(WithInstanceId(Autocomplete)))
    in WithSpokenMessages(WithInstanceId(Autocomplete)) (created by BlockEditorAutocomplete)
    in BlockEditorAutocomplete (created by ForwardRef(RichText))
    in ForwardRef(RichText) (created by ForwardRef(RichTextWrapper))
    in ForwardRef(RichTextWrapper) (created by HeadingEdit)
    in HeadingEdit (created by Edit)
    in Edit (created by WithToolbarControls(Edit))
    in WithToolbarControls(Edit) (created by WithInspectorControl(WithToolbarControls(Edit)))
    in WithInspectorControl(WithToolbarControls(Edit)) (created by WithInspectorControl(WithInspectorControl(WithToolbarControls(Edit))))
    in WithInspectorControl(WithInspectorControl(WithToolbarControls(Edit))) (created by WithToolbarControls(WithInspectorControl(WithInspectorControl(WithToolbarControls(Edit)))))
    in WithToolbarControls(WithInspectorControl(WithInspectorControl(WithToolbarControls(Edit))))
    in Unknown (created by WithDispatch(Component))
    in WithDispatch(Component)
    in Unknown (created by WithMultipleValidation(WithToolbarControls(WithInspectorControl(WithInspectorControl(WithToolbarControls(Edit))))))
    in WithMultipleValidation(WithToolbarControls(WithInspectorControl(WithInspectorControl(WithToolbarControls(Edit))))) (created by WithFilters(Edit))
    in WithFilters(Edit) (created by BlockEdit)
    in BlockEdit (created by BlockListBlock)
    in BlockCrashBoundary (created by BlockListBlock)
    in BlockListBlock (created by (BlockListBlock))
    in (BlockListBlock) (created by ((BlockListBlock)))
    in ((BlockListBlock)) (created by WithFontSizeInlineStyles(((BlockListBlock))))
    in WithFontSizeInlineStyles(((BlockListBlock))) (created by WithFilters(BlockListBlock))
    in WithFilters(BlockListBlock) (created by IfCondition(WithFilters(BlockListBlock)))
    in IfCondition(WithFilters(BlockListBlock)) (created by WithDispatch(IfCondition(WithFilters(BlockListBlock))))
    in WithDispatch(IfCondition(WithFilters(BlockListBlock)))
    in Unknown (created by WithSelect(WithDispatch(IfCondition(WithFilters(BlockListBlock)))))
    in WithSelect(WithDispatch(IfCondition(WithFilters(BlockListBlock))))
    in Unknown (created by Pure(WithViewportMatch(WithSelect(WithDispatch(IfCondition(WithFilters(BlockListBlock)))))))
    in Pure(WithViewportMatch(WithSelect(WithDispatch(IfCondition(WithFilters(BlockListBlock)))))) (created by ForwardRef(BlockList))
    in div (created by ForwardRef(RootContainer))
    in div (created by InsertionPoint)
    in InsertionPoint (created by ForwardRef(RootContainer))
    in ForwardRef(RootContainer) (created by ForwardRef(BlockList))
    in ForwardRef(BlockList) (created by ForwardRef)
    in ForwardRef
    in div (created by FormStylingBackground)
    in FormStylingBackground
    in div (created by ObserveTyping)
    in ObserveTyping (created by WithSafeTimeout(ObserveTyping))
    in WithSafeTimeout(ObserveTyping)
    in div (created by WritingFlow)
    in div (created by WritingFlow)
    in WritingFlow
    in div
    in div (created by BlockSelectionClearer)
    in BlockSelectionClearer (created by WithDispatch(BlockSelectionClearer))
    in WithDispatch(BlockSelectionClearer)
    in Unknown (created by WithSelect(WithDispatch(BlockSelectionClearer)))
    in WithSelect(WithDispatch(BlockSelectionClearer))
    in div
    in BlockEditorProvider
    in Unknown (created by Context.Consumer)
    in WithRegistryProvider(BlockEditorProvider)
    in div
    in div
    in SlotFillProvider (created by SlotFillProvider)
    in SlotFillProvider
    in div (created by DropZoneProvider)
    in DropZoneProvider
    in Unknown (created by App)
    in App
    in StrictMode
```

File: assets/js/src/form_editor/store/reducers/toggle_sidebar_panel.jsx (hunk 1 above)
Comment (pavel-mailpoet) [reply assets/js/src/form_editor/store/reducers/toggle_sidebar_panel.jsx:24]: Because the component doesn't return string any more, it returns a boolean value.

**Comment by pavel-mailpoet** [comment]
@costasovo I fixed the custom text problem. The react warning is, unfortunately, a bug in Gutenberg and I cannot fix it.

**Review by costasovo — APPROVED** [review]

**Review by veljkho — APPROVED** [review]
